addons/website_sale/static/src/js/website_sale_validate.js

```diff
@@ -12,7 +12,7 @@ $(document).ready(function () {
         if (! order_node || order_node.data('order-id') === undefined) {
             return;
         }
-        var order_id = order_node.data('order-id');
+        var order_id = parseInt(order_node.data('order-id'));
         return ajax.jsonRpc('/shop/payment/get_status/' + order_id, 'call', {
```

Comment:
warning: if order-id is invalid this will call `/shop/payment/get_status/NaN`, I don't know if it's better than the old incorrect URL.